### ISSUE TITLE
FFTW single-precision library outputs incorrect DFT when using SIMD optimizations

### ISSUE THREAD

**Issue by poxxo** [FFTW/fftw3#204]
With FFTW 3.3.8, I often get incorrect results when I use the float (single-precision) version of the libraries, configured with the `--enable-generic-simd128` or `--enable-generic-simd256` options (and `--enable-float` for `float` support) and built from source. I have tested this with gcc versions 10.1.0 (MinGW-w64), 7.5.0 (Windows Subsystem for Linux), 9.3.0 (Ubuntu) and clang 10.0.0 (Ubuntu), with Intel i7-4720HQ and i7-8550U CPUs. I get the same incorrect DFT in all cases. I also get the same errors when I use the pre-built FFTW package for MinGW-w64, downloaded via pacman in MSYS2.

As a simple test, I'm using an input vector of 1's. The first element of the expected DFT should be equal to the length of the input vector, with all the other elements zero.

Here is a simple test program to demonstrate the issue:
main.c
```
#include <stdio.h>
#include <fftw3.h>

int main()
{
    fftwf_complex *in, *out;
    fftwf_plan p;
    int N = 21;
    int i;

    in = fftwf_malloc(sizeof(fftwf_complex) * N);
    out = fftwf_malloc(sizeof(fftwf_complex) * N);
    p = fftwf_plan_dft_1d(N, in, out, FFTW_FORWARD, FFTW_ESTIMATE);

    for (i = 0; i < N; i++) {
        in[i][0] = 1.0f;
        in[i][1] = 0.0f;
    }
    fftwf_execute(p);
	
    for (i = 0; i < N; i++) 
	printf("%d: %8.5g\t + j %8.5g\n", i, out[i][0], out[i][1]);

    fftwf_destroy_plan(p);
    fftwf_free(in);     fftwf_free(out);

    return 0;
}
```
I build it with ```gcc -o main main.c -lfftw3f -lm```. The output is as follows:
```
0:       21      + j    -7.87
1:        0      + j        0
2:        0      + j        0
3:  -5.2972      + j      1.9
4:        0      + j        0
5:        0      + j        0
6:   -1.862      + j     1.08
7:        0      + j        0
8:        0      + j        0
9: -0.52584      + j    0.956
10:        0     + j        0
11:        0     + j        0
12:  0.52584     + j    0.956
13:        0     + j        0
14:        0     + j        0
15:    1.862     + j     1.08
16:        0     + j        0
17:        0     + j        0
18:   5.2972     + j      1.9
19:        0     + j        0
20:        0     + j        0
```
The 0'th element should be 21, with the rest equal to 0 (or at least close).

**Comment by matteo-frigo** [comment]
Thanks for the bug report.  I believe the bug to be fixed in d832ca0e, can you confirm?

**Comment by poxxo** [comment]
[d832ca0](https://github.com/FFTW/fftw3/commit/d832ca0e2f286650278563c6887c4f235785e606) does indeed appear to have fixed this, tested with gcc and Clang. Thanks for your hard work!
